### ISSUE TITLE
Feature Request (and bug?): validate posts (and comments?) before they are sent

### ISSUE THREAD

**Issue by janboddez** [Automattic/wordpress-activitypub#631]
### What

As [mentioned elsewhere](https://github.com/Automattic/wordpress-activitypub/issues/628#issuecomment-1875066820), I'm not sure posts are sufficiently validated before being federated. ~~(I may be wrong, though!)~~

In another plugin I created, I also allow scheduling of HTTP POST requests in a very similar manner, and I basically check (once more!) that a post is still published, isn't password protected, and so on, right before it is being sent out.

~~Now, I may be overthinking this, or it may already be in one of the `Transformer` classes that I've missed. In that case, feel free to close this issue.~~

### Why

If users use _actual cron_ and have WP-Cron disabled as (used to be?) suggested by the plugin, HTTP POST requests (to followers' outboxes) could be delayed by as much as, e.g., 5 or even 10 minutes. (And even if the cron job runs every minute or if they do rely on WP-Cron, there might similarly be race conditions or something.)

A post may have changed sufficiently between them clicking "Publish" for the very first time and it being sent out. Seems to me that all these checks (plus, possibly, checks for mentions/tags/etc.) have to be re-run (or even delayed to the point) when the post is _actually_ going to get sent.

(Also, if you ever wanted to offer a checkbox or so to opt-out and it was going to be a _classic_ meta box [and not Gutenberg-powered], there's this "weird" behavior where Gutenberg will _first_ publish the post [triggering the cron event to be scheduled] but only _then_ allow processing of the meta box's values.)

TL;DR: "race" conditions.

### How

Same way you do, e.g.: https://github.com/Automattic/wordpress-activitypub/blob/4c297acfd454a13260900ff0f97a9e3f425e960c/includes/class-activity-dispatcher.php#L74

Except run additional checks.
